doc/releases/release-notes-4.0.rst

```diff
@@ -410,6 +551,8 @@ Drivers and Sensors
 
 * I3C
 
+  * Moved clocking code out of the NXP I3C driver
```

Comment:
not sure if this needs to go in the release notes as it isn't exposed to users, it's basically just a refactor of where some code was in the tree